serde/src/ser/mod.rs

```diff
@@ -124,6 +124,14 @@ pub trait Serializer {
         self.visit_unit()
     }
 
+    #[inline]
+    fn visit_enum_simple<T>(&mut self,
+                            _name: &str,
+                            _variant: &str,
+                            _value: T,
+                            ) -> Result<(), Self::Error>
+        where T: Serialize;
```

Comment:
same here, instead of requiring this to be implemented, it could call visit_seq on some locally generated sequence visitor with a single element